Adding Great Britain open mag dataset

nshea3 opened the pull request:
<!-- Please describe changes proposed and WHY you made them. If unsure, open an issue first to discuss the idea. -->

Hi,

I opened up the data and threw together a quick scatterplot. 

![image](https://user-images.githubusercontent.com/25097799/66348424-cef2f900-e956-11e9-815f-c7b8c65a9c8e.png)

The archive `gb-magnetic.csv.xz` is 2.7 Mb compressed.

I have converted the coordinates from OSGB36 to WGS84 using geopandas. 

I have zipped the csv, added the archive and the SHA256 hash to the Pooch registry at `harmonica/datasets/registry.txt`.

I created a loading function (`fetch_gb_magnetic`, includes docstring) and a gallery example (`gb_magnetic.py`)

I have also added in unit testing in `test_sample_data.py`.

<!-- If fixing an issue, put the issue number after the # below (no spaces). Github will automatically close it when this gets merged. -->
Fixes #101 


**Reminders**

- [x] Run `make format` and `make check` to make sure the code follows the style guide.
- [x] Add tests for new features or tests that would have caught the bug that you're fixing.
- [x] Add new public functions/methods/classes to `doc/api/index.rst`.
- [x] Write detailed docstrings for all functions/methods.
- [x] If adding new functionality, add an example to the docstring, gallery, and/or tutorials.

leouieda commented:
Hi @nshea3 thank you so much for the contribution :1st_place_medal: ! Me or @santisoler will review the code shortly.

leouieda commented on harmonica/datasets/sample_data.py:
```diff
@@ -140,6 +140,30 @@ def fetch_rio_magnetic():
+    These data are a complete airborne survey of the entire United Kingdom
+    conducted between 1955 and 1965. The data are made available by the
+    British Geological Survey (BGS) through their `geophysical data portal
+    <https://www.bgs.ac.uk/products/geophysics/aeromagneticRegional.html>`__.
```
```suggestion
    <https://www.bgs.ac.uk/products/geophysics/aeromagneticRegional.html>`__.
    
    License: `Open Government License 
    <http://www.nationalarchives.gov.uk/doc/open-government-licence/version/3/>`__
```

leouieda replied:
We're trying to standardize this in our sample datasets. 

@santisoler what do you think of having this `License:` section in the docstring?

santisoler replied:
Sure! I think we should add the license both on the gallery example and on the docstring of the function that fetches the dataset. Besides, I think adding the name of the license and a link to the original license text would be enough.

leouieda commented on data/examples/gb_magnetic.py:
```diff
@@ -0,0 +1,52 @@
+print(data)
+
+# Plot the observations in a Mercator map using Cartopy
+fig = plt.figure(figsize=(10, 6))
```
```suggestion
fig = plt.figure(figsize=(7.5, 10))
```

leouieda commented on data/examples/gb_magnetic.py:
```diff
@@ -0,0 +1,52 @@
+Total Field Magnetic Anomaly from Great Britain
+================================================
+
+These data are a complete airborne survey of the entire United Kingdom
```
```suggestion
These data are a complete airborne survey of the entire Great Britain
```

leouieda replied:
Northern Ireland was left out of this one :slightly_frowning_face:

leouieda commented on harmonica/datasets/sample_data.py:
```diff
@@ -140,6 +140,30 @@ def fetch_rio_magnetic():
 
+def fetch_gb_magnetic():
+    """
+    Fetch total-field magnetic anomaly data of the United Kingdom.
```
```suggestion
    Fetch total-field magnetic anomaly data of Great Britain.
```

leouieda commented on harmonica/datasets/sample_data.py:
```diff
@@ -140,6 +140,30 @@ def fetch_rio_magnetic():
+    """
+    Fetch total-field magnetic anomaly data of the United Kingdom.
+
+    These data are a complete airborne survey of the entire United Kingdom
```
```suggestion
    These data are a complete airborne survey of the entire Great Britain
```

leouieda commented:
Please let me know if you have any problems or need any help. And of course, feel free to disagree with any of my suggestions :+1:

nshea3 commented on data/examples/gb_magnetic.py:
```diff
@@ -0,0 +1,52 @@
+maxabs = vd.maxabs(data.MAG_IGRF90)
+tmp = ax.scatter(
+    data.LONGITUDE,
+    data.LATITUDE,
```
When you say that it could be done in the code instead of the data file, does that mean that I should rename them when the file is opened in this example script?
Or would it be better that I rename the columns as part of preprocessing the data?

leouieda replied:
I meant rename them when loading the data in `fetch_gb_magnetic` so the user never sees the original names. Or you could rename them in the original data file and keep the code as is. Whichever is easier for you, actually.

nshea3 commented on harmonica/datasets/sample_data.py:
```diff
@@ -140,6 +140,30 @@ def fetch_rio_magnetic():
+
+    The columns of the data table are longitude, latitude, total-field
+    magnetic anomaly (nanoTesla), observation height relative to Ordnance Survey
+    datum (in meters), survey area, and line number and line segment for each data point.
```
I will look into this one.

nshea3 commented:
Thank you for the feedback! 

That map looks a lot better, thanks for the tip. I'm on board with all the naming changes as well.

In terms of work to be done, I will fill in some of the details on the datum and the conversion to WGS84. I will also look into converting the elevations.

leouieda commented:
Nice, thanks @nshea3! Keep us informed of your progress and don't hesitate to ask if you need any guidance (or want to discuss something). The suggestions I left here can be commited to your branch directly by clicking on the "Commit suggestion" box if you don't want to have to copy them offline.

nshea3 commented:
Hi,
Sorry I was not able to get to this sooner. 
I changed `gb` to `britain` in all of the function, method, and file names. 
I added a few more lines to the data processing script to change the datum for the terrain elevation. I added this value to the average terrain clearance height to get the altitude in meters. Additionally the csv now has lowercase column names survey_area, line-number-segment, longitude, latitude, altitude_m, and total_field_anomaly_nt.
Let me know how the changes look! 
Nick

leouieda commented:
@nshea3 no worries at all! I know how tough it is to keep up with development (I'm usually the one behind). Changes look great! I merged in the master branch and will merge this PR as soon as all checks pass. 

Thanks for the great work! :1st_place_medal: I really appreciate your time and dedication.

santisoler commented:
Thanks @nshea3 for taking the time and job to implement this! I'm already needing this dataset for my current research, so I'm glad to see this already on master! Happy to see new contributors on Harmonica! :1st_place_medal:

nshea3 commented:
Thank you both for your help! I'm glad to hear the the dataset is getting use. I look forward to contributing again soon!
